thirdparty/runtime_environments.go

```diff
@@ -58,3 +59,53 @@ func (c *RuntimeEnvironmentsClient) getImageNames(ctx context.Context) ([]string
 	}
 	return filteredImages, nil
 }
+
+// ImageDiffOptions represents the arguments for getImageDiff. BeforeAMI is the starting AMI, and AfterAMI is the ending AMI.
+type ImageDiffOptions struct {
+	BeforeAMI string
+	AfterAMI  string
+}
+
+// ImageDiffChange represents a change between two AMIs.
+type ImageDiffChange struct {
+	Name    string
+	Manager string
+	Type    string
+	Removed string
+	Added   string
+}
+
+// getImageDiff returns a list of package and toolchain changes that occurred between the provided AMIs.
+func (c *RuntimeEnvironmentsClient) getImageDiff(ctx context.Context, opts ImageDiffOptions) ([]ImageDiffChange, error) {
+	params := url.Values{}
+	params.Set("ami", opts.BeforeAMI)
+	params.Set("ami2", opts.AfterAMI)
+	params.Set("limit", "1000000000") // Artificial limit set high because API has default limit of 10.
+	apiURL := fmt.Sprintf("%s/rest/api/v1/imageDiffs?%s", c.BaseURL, params.Encode())
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Api-Key", c.APIKey)
+	resp, err := c.Client.Do(request)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		msg, _ := io.ReadAll(resp.Body)
+		return nil, errors.Errorf("HTTP request returned unexpected status '%s': %s", resp.Status, string(msg))
+	}
+	changes := []ImageDiffChange{}
+	if err := gimlet.GetJSON(resp.Body, &changes); err != nil {
+		return nil, errors.Wrap(err, "decoding http body")
+	}
+	filteredChanges := []ImageDiffChange{}
+	for _, c := range changes {
+		if c.Type == "Packages" || c.Type == "Toolchains" {
```

Comment:
lets make these a const somewhere so we're not just checking for a random string